li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaInstrumentedProducerImpl.java

```diff
@@ -287,6 +308,58 @@ public void flush() {
     }
   }
 
+  /**
+   * A temporary hack that implements the bounded flush API defined in LiKafkaProducer.
+   * TODO: remove when the bounded flush is added to upstream.
+   *
+   * This method will flush all the message buffered in producer. The call blocks until timeout.
+   * If the underlying producer doesn't support a bounded flush, it will invoke the {@link #flush()}.
+   */
+  public void flush(long timeout, TimeUnit timeUnit) {
+    verifyOpen();
+
+    delegateLock.readLock().lock();
+    try {
+      boolean useSeparateThreadForFlush = false;
+      if (boundedFlushMethod != null) {
+        try {
+          boundedFlushMethod.invoke(delegate, timeout, timeUnit);
+        } catch (IllegalAccessException | InvocationTargetException e) {
+          LOG.trace("Underlying producer does not support bounded flush!", e);
```

Comment:
this should never happen as we look for the method as construction time. as such, this should be an IllegalStateException, and not swallowed.

Reply:
done